### ISSUE TITLE
MPCD AT gives NaN particle positions

### ISSUE THREAD

**Issue by mphoward** [glotzerlab/hoomd-blue#372]
## Description

The MPCD example script crashes with NaN particle positions in single-precision builds on both the CPU & GPU when SRD is replaced by AT. The error is not present in double-precision builds or when the seed is changed to a different value.

I suspect the root cause is in the Box-Muller transformation used to generate normal random variates:

https://github.com/glotzerlab/hoomd-blue/blob/707e5163b0e15e5d2bae1f992f985c7d97ebef32/hoomd/mpcd/RandomNumbers.h#L93-L94

Likely, for some combination of RNG hash parameters, either `-2*log(u1)` is negative or `u1` is exactly zero, giving a NaN from that math function call. If this is true, an appropriate fix would be to (1) replace `u1` by `1.0f-u1` so that the variate is (0,1] rather than [0,1), and then (2) repeat the draw in a do loop until `-2*log(u1)` is nonnegative (see lines 50-54 for sphere point picking). The other caused could be another bug in Saru, so fingers crossed that it is the former rather than the latter, as this was very hard to track down last time.

## Script

```python
import hoomd
hoomd.context.initialize()
from hoomd import mpcd

box = hoomd.data.boxdim(L=10.)
hoomd.init.read_snapshot(hoomd.data.make_snapshot(N=0, box=box))

s = mpcd.init.make_random(N=int(5*box.get_volume()), kT=1.0, seed=42)

mpcd.integrator(dt=0.1)
mpcd.stream.bulk(period=1)
# crashes
mpcd.collide.at(seed=42, period=1, kT=1.0)
# doesn't crash
#mpcd.collide.srd(seed=42, angle=130, period=1, kT=1.0)

hoomd.run(2000)
```

## Output

```
**ERROR**: MPCD particle 3692 has position NaN
Traceback (most recent call last):
  File "test_at.py", line 17, in <module>
    hoomd.run(2000)
  File "/home/mphoward/packages/glotzerlab/hoomd-blue/v2.5.1/nompi/single/hoomd/__init__.py", line 201, in run
    context.current.system.run(int(tsteps), callback_period, callback, limit_hours, int(limit_multiple));
RuntimeError: Error computing cell list
```

## Expected output

It should have run to completion.

## Configuration

Platform:
- Linux
- GPU
- CPU

Installation method:
- Compiled from source

Versions:
- Python version: 3.4.3
- **HOOMD-blue** version: 2.5.1

## Developer

I will fix this.

**Comment by mphoward** [comment]
As suspected, there are certain hashes that give `inf` values. One example is `tag = 1363`, `timestep=316`, and `seed = 42`.

**Comment by mphoward** [comment]
The first time, this was just because the RNG drew exactly 0.0. This was easy to fix.

I found another hash that also fails `(309, 1313, 42)`. This seems to instead originate from an underlying issue with Saru... *sigh*

**Comment by joaander** [comment]
Fixed in v2.5.2
